content/posts/2020/12/libplanet-0.10/index.eng.md

```diff
@@ -0,0 +1,144 @@
+
+And to help with debugging, the newly added `planet mpt` command makes it easy to compare the status between blocks, or import the status from a specific block. Please refer to the `planet mpt --help` option for instructions on how to use it.
+
+Separate State API for Assets
+------------------------------
+
+So far, in-game goods have been treated in the same way as in-game states when creating games with Libplanet. For example, NCG (〈Nine Chronicles〉 Gold) is implemented as an integer data type value. Although such goods should never be replicated or destroyed carelessly, implementing them in a data type that is free of proprietary computation and doesn't necessarily hold the characteristics of the goods is prone to bugs.
+
+For instance, when money is transferred, the sender's balance should be reduced and the recipient's balance should be increased. However, bugs that skip updating either the sender's or the recipient's balance could take place. Also, bugs that distribute the money to multiple people and bugs that unintentionally delete remaining amounts after division are possible.
+
+And even more critically, when a programmer carelessly writes a code that just adds money to players' balance as a game reward, this could affect the game economy as a whole and is practically like casting money privately.
+
+To prevent these issues early on, this version has a separate state API for primarily dealing with assets. The [`BlockChain<T>.GetBalance()`] and [`IAccountStateDelta.GetBalance()`] methods are created alongside the existing [`BlockChain<T>.GetState()`] and [`IAccountStateDelta.GetState()`] methods. And in addition to the [`IAccountStateDelta.SetState()`] method which can freely update states, [`IAccountStateDelta.TransferAsset()`] method for transferring assets and [`IAccountStateDelta.MintAsset()`] method for minting assets have been created.
+
+Also, instead of using .NET's built-in integer data type, you need to use the new [`FungibleAssetValue`] data type added to Libplanet. `FungibleAssetValue` basically looks like a [`BigInteger`], but there are some differences.
+
+1. In division, the remaining values are never implicitly discarded and are always treated explicitly.
+Therefore, instead of implementing the division operator(`/`), only the [`DivRem()`] method is implemented.
+2. It supports minor currency units, such as dollar–cent, and limits the number of digits in the lower unit.
+3. It preserves the unit of currency of each value so that different currencies do not mix.
+To implement No.3 above, the data type [`Currency`] is created to define the monetary unit. Its property includes the name of the currency unit, ticker symbol, and the number of digits in the lower unit.
+
+Currently, asset state API only supports fungible assets such as game money, but it will also support Non-Fungible Assets([<abbr title="non-fungible token">NFT</abbr>][NFT]) such as game items in the future.
```

Comment:
```suggestion
Currently, asset state API only supports fungible assets such as game money, but it will also support [<abbr title="non-fungible token">NFT</abbr>][NFT] such as game items in the future.
```